src/test/kotlin/org/rust/ide/inspections/RsBinaryOpOrBinaryOpAssignInspectionTest.kt

```diff
@@ -0,0 +1,243 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections
+
+import org.rust.ProjectDescriptor
+import org.rust.WithStdlibRustProjectDescriptor
+
+class RsBinaryOpOrBinaryOpAssignInspectionTest : RsInspectionsTestBase(RsBinaryOpOrBinaryOpAssignInspection::class) {
+
+    fun `test bit and op assign`() = checkErrors("""
+        struct Test(i32);
+
+        fn foo(a: Test, b: Test) {
+            <error descr="binary assignment operation `&=` cannot be applied to type `Test` [E0368]">a &= b</error>;
+        }
+    """)
+
+    fun `test bit or op assign`() = checkErrors("""
+        struct Test(i32);
+
+        fn foo(a: Test, b: Test) {
+            <error descr="binary assignment operation `|=` cannot be applied to type `Test` [E0368]">a |= b</error>;
+        }
+    """)
+
+    fun `test bit xor op assign`() = checkErrors("""
+        struct Test(i32);
+
+        fn foo(a: Test, b: Test) {
+            <error descr="binary assignment operation `^=` cannot be applied to type `Test` [E0368]">a ^= b</error>;
+        }
+    """)
+
+    fun `test shl op assign`() = checkErrors("""
+        struct Test(i32);
+
+        fn foo(a: Test, b: Test) {
+            <error descr="binary assignment operation `<<=` cannot be applied to type `Test` [E0368]">a <<= b</error>;
```

Comment:
Bug:
* desc truncated to 
```
binary assignment operation `E0368]
```

Reply:
@Undin Can you look at this bug?

Reply:
It seems that I found the problem.

https://github.com/JetBrains/intellij-community/blob/3ab71c1c582869742dd2da068c1834963798a7b1/platform/lang-impl/src/com/intellij/codeInsight/daemon/impl/LocalInspectionsPass.java#L546

The problem appears on the regular expression of this line.

Reply:
Found it:
#2082
and
https://youtrack.jetbrains.com/issue/IDEA-186991